lib/logstash/event.rb

```diff
@@ -295,11 +295,7 @@ def init_timestamp(o)
 
   public
   def to_hash_with_metadata
-    if @metadata.nil?
-      to_hash
-    else
-      to_hash.merge(METADATA => @metadata)
-    end
+    to_hash.merge(METADATA => @metadata)
```

Comment:
I just verified everywhere:
- it seems that `to_json_with_metadata` is not used anywhere.
- `to_hash_with_metada` is used in rubydebug codec, multiline filter and couchdb_changes input, in all cases it make no difference.

So I'd say we do what's fastest: either checking for `empty?` or doing a hash `merge` I would assume checking for `empty?` is fastest then `merge`...?